core/src/main/java/org/mule/runtime/core/internal/processor/chain/SubflowMessageProcessorChainBuilder.java

```diff
@@ -36,7 +38,7 @@ protected MessageProcessorChain createInterceptingChain(Processor head, List<Pro
   /**
    * Generates message processor identifiers specific for subflows.
    */
-  static class SubFlowMessageProcessorChain extends DefaultMessageProcessorChain {
+  static class SubFlowMessageProcessorChain extends DefaultMessageProcessorChain implements ProcessorChainPrototype {
```

Comment:
We should really replace `SubFlowMessageProcessorChain` with a `SubFlowMessageProcessorChainBuilder` instead of leaving it as a chain and then converting into builder and back into chain again elsewhere.